clang/lib/Sema/SemaDecl.cpp

```diff
@@ -11792,6 +11792,32 @@ static bool CheckMultiVersionFunction(Sema &S, FunctionDecl *NewFD,
                                          OldDecl, Previous);
 }
 
+static void CheckFunctionDeclarationAttributesUsage(Sema &S,
+                                                    FunctionDecl *NewFD) {
+  const bool is_pure = NewFD->hasAttr<PureAttr>();
+  const bool is_const = NewFD->hasAttr<ConstAttr>();
+
+  if (is_pure && is_const) {
+    S.Diag(NewFD->getLocation(), diag::warn_const_attr_with_pure_attr);
+    NewFD->dropAttr<PureAttr>();
```

Comment:
Why drop 'pure'?  It is more constrained than const, right?  it would seem that keeping 'pure' around and dropping const would be more consistent.

Reply:
> The const attribute imposes greater restrictions on a function’s definition than the similar pure attribute. 

https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html

Reply:
Got it, had those reversed in my head.  Thanks for the clarification.